pkg/translate/translateValue_test.go

```diff
@@ -0,0 +1,389 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package translate
+
+import (
+	"testing"
+
+	"github.com/ghodss/yaml"
+	"github.com/kr/pretty"
+
+	"istio.io/operator/pkg/apis/istio/v1alpha2"
+	"istio.io/operator/pkg/util"
+	"istio.io/operator/pkg/version"
+)
+
+func TestValueToProto(t *testing.T) {
+
+	tests := []struct {
+		desc      string
+		valueYAML string
+		want      string
+		wantErr   string
+	}{
+		{
+			desc: "All Enabled",
+			valueYAML: `
+certManager:
+  enabled: true
+galley:
+  enabled: true
+global:
+  hub: docker.io/istio
+  istioNamespace: istio-system
+  policyNamespace: istio-policy
+  tag: 1.2.3
+  telemetryNamespace: istio-telemetry
+mixer:
+  policy:
+    enabled: true
+  telemetry:
+    enabled: true
+pilot:
+  enabled: true
+nodeAgent:
+  enabled: true
+gateways:
+  enabled: true
+  istio-ingressgateway:
+    resources:
+      requests:
+        cpu: 1000m
+        memory: 1G
+    enabled: true
+sidecarInjectorWebhook:
+  enabled: false
+`,
+			want: `
+hub: docker.io/istio
+tag: 1.2.3
+default_namespace: istio-system
+telemetry:
+  components:
+    namespace: istio-telemetry
+    telemetry:
+      common:
+        enabled:
+          value: true
+  enabled:
+    value: true
+policy:
+  components:
+    namespace: istio-policy
+    policy:
+      common:
+        enabled:
+          value: true
+  enabled:
+    value: true
+config_management:
+  components:
+    galley:
+      common:
+        enabled:
+          value: true
+  enabled:
+    value: true 
+security:
+  components:
+    namespace: istio-system
+    cert_manager:
+      common:
+        enabled:
+          value: true
+    node_agent:
+      common:
+        enabled:
+          value: true
+    citadel:
+      common:
+        enabled: {}
+  enabled:
+    value: true
+traffic_management:
+   components:
+     pilot:
+       common:
+         enabled: 
+           value: true
+   enabled: 
+     value: true
+auto_injection:
+  components:
+    injector:
+      common:
+        enabled: {}
+  enabled: {}
+gateways:
+  components:
+    ingress_gateway:
+          common:
+           enabled:
+             value: true
+           k8s:
+             resources:
+               requests:
+                 cpu: 1000m
+                 memory: 1G 
+    egress_gateway:
+          common:
+            enabled: {}
+  enabled:
+    value: true
+`,
+		},
+		{
+			desc: "Some components Disabled",
+			valueYAML: `
+galley:
+  enabled: false
+pilot:
+  enabled: true
+global:
+  hub: docker.io/istio
+  istioNamespace: istio-system
+  policyNamespace: istio-policy
+  tag: 1.2.3
+  telemetryNamespace: istio-telemetry
+mixer:
+  policy:
+    enabled: true
+telemetry:
+    enabled: false
+`,
+			want: `
+hub: docker.io/istio
+tag: 1.2.3
+default_namespace: istio-system
+telemetry:
+ components:
+   namespace: istio-telemetry
+   telemetry:
+     common:
+       enabled: {}
```

Comment:
shouldn't this be false? There seems to be something off with the type in this field.

Reply:
change to use jsonpb marshaller